experimental/dds/tree2/src/feature-libraries/contextuallyTyped.ts

```diff
@@ -467,13 +510,33 @@ export function applyTypesFromContext(
 		const fields: Map<FieldKey, MapTree[]> = new Map();
 		for (const key of fieldKeysFromData(data)) {
 			assert(!fields.has(key), 0x6b3 /* Keys should not be duplicated */);
-			const childSchema = getFieldSchema(key, schemaData, schema);
-			const children = applyFieldTypesFromContext(schemaData, childSchema, data[key]);
+			const childSchema = getFieldSchema(key, context.schema, schema);
+			const children = applyFieldTypesFromContext(context, childSchema, data[key]);
+
 			if (children.length > 0) {
 				fields.set(key, children);
 			}
 		}
 
+		// Generate and set data for global keys that have value fields.
+		if (context.schema.globalFieldSchema !== undefined) {
+			for (const [key] of context.schema.globalFieldSchema) {
```

Comment:
rather than looping through all possible global fields in the document, I think it would be better from a performance perspective to loop through all the fields in the schema and find any required ones which are missing in the data.

Reply:
Just to clarify, is the `schema` that we should loop through referring to the one defined in the context object (`context.schema`), or the `schema` from `const schema = lookupTreeSchema(schemaData, type);`?